testing/prow/gpu-operator.sh

```diff
@@ -109,30 +109,57 @@ prepare_cluster_for_gpu_operator() {
 
 collect_must_gather() {
     run_in_sub_shell() {
-        echo "Running gpu-operator_gather ..."
-        /usr/bin/gpu-operator_gather &> /dev/null
-
-        # extract ARTIFACT_EXTRA_LOGS_DIR from 'source toolbox/_common.sh' without sourcing it directly
-
-        export TOOLBOX_SCRIPT_NAME=toolbox/gpu-operator/must-gather.sh
-        COMMON_SH=$(
-            bash -c 'source toolbox/_common.sh;
-                     echo "8<--8<--8<--";
-                     # only evaluate these variables from _common.sh
-                     env | egrep "(^ARTIFACT_EXTRA_LOGS_DIR=)"'
-                 )
-        ENV=$(echo "$COMMON_SH" | tac | sed '/8<--8<--8<--/Q' | tac) # keep only what's after the 8<--
-        eval $ENV
+        echo "Running the GPU Operator must-gather image ..."
+        OPERATOR_IMAGE=$(oc get pods -A -lapp=gpu-operator -o=jsonpath='{range .items[*]}{range .spec.containers[*]}{.image}{end}{end}' || true)
+
+        TMP_DIR="$(mktemp -d -t gpu-operator_XXXX)"
+
+        if [[ "$OPERATOR_IMAGE" ]]; then
+            echo "Operator image: $OPERATOR_IMAGE"
+
+            oc adm must-gather --image="$OPERATOR_IMAGE" --dest-dir="${TMP_DIR}" &> /dev/null
+
+            # ${TMP_DIR}/<image>/ should contain the file generated by
+            # the must-gather script. If this is empty, there wasn't a
+            # must-gather script in the image!
+            if [[ "$(ls "${TMP_DIR}"/*/* 2>/dev/null | wc -l)" == 0 ]]; then
+                echo "GPU Operator image failed to must-gather anything ..."
+            else
+                img_dirname=$(dirname "$(ls "${TMP_DIR}"/*/* | head -1)")
+                mv "$img_dirname"/* $TMP_DIR
+                rmdir "$img_dirname"
+
+                # extract ARTIFACT_EXTRA_LOGS_DIR from 'source toolbox/_common.sh' without sourcing it directly
+                export TOOLBOX_SCRIPT_NAME=toolbox/gpu-operator/must-gather.sh
+                COMMON_SH=$(
+                    bash -c 'source toolbox/_common.sh;
```

Comment:
`bash -c` is redundant:
![image](https://user-images.githubusercontent.com/10882062/145194484-7aff7832-7660-4e30-9f02-a2aab192068f.png)

Reply:
good catch